docs/preview/features/ps/keyvault.md

```diff
@@ -0,0 +1,38 @@
+---
+title: "Scripts related to interacting with Azure Key Vault"
+layout: default
+---
+
+# Azure Key Vault scripting
+
+## Installation
+
+To have access to the following features, you have to import the module:
+
+```powershell
+PS> Import-Module -Name Arcus.Scripting.PS.KeyVault
+```
+
+## Get all access policies of a key vault
+
+Lists the current available access policies of the Azure key vault resource into a new variable availble at runtime during a Azure DevOps build.
+
+| Parameter            | Mandatory                                     | Description                                                                  |
+| -------------------- | --------------------------------------------- | ---------------------------------------------------------------------------- |
+| `KeyVaultName`       | yes                                           | The name of the key vault from which the access policies are to be retrieved |
+| `ResourceGroupName`  | no                                            | The resource group containing the key vault                                  |
+| `OutputVariableName` | no (default: `Infra.KeyVault.AccessPolicies`) | The name of the variable to be added to DevOps-pipeline variables at runtime |
+
+**Example**
+
+```powershell
+PS> Get-KeyVaultAccessPolicies -KeyVaultName "my-key-vault"
+# Output: #vso[task.setvariable variable=Infra.KeyVault.AccessPolicies]{list: [{tenantId: ...,permissions: ...}]}
```

Comment:
This is confusing - Does the command set this as an output variable or also write it? Shouldn't it just return the payload and you can capture it similar to this?

```powershell
$acls = Get-KeyVaultAccessPolicies -KeyVaultName "my-key-vault" -ResourceGroupName "my-resouce-group" -OutputVariableName "My.KeyVault.AccessPolicies"
```

And then do this:
```powershell
#vso[task.setvariable variable=My.KeyVault.AccessPolicies] $acls
```

Because it feels like we are mixing two things here.

Reply:
This script was ment to capture the current access policies and set it as an pipeline variable to be passed along to an ARM-template.

Either we rename this command  and create to second:
- Get-KeyVaultAccessPolicies: actually outputs the value
- Write-KeyVaultAccessPoliciesToDevOpsVariable: adds the variable
(or some better naming :) )

We could expect the user to use `#vso[task.setvariable variable=My.KeyVault.AccessPolicies] $acls` themselves, but this would add some additional "complexity", while this should be all about making it easier.

Reply:
What if that `Write-KeyVaultAccessPoliciesToDevOpsVariable` simply becomes `Set-DevOpsVariable` instead? That way you can re-use it for others as well.

Reply:
We can definitely use a script such as `Set-DevOpsVariable` (#44 has been created for this)

Makes sense, we'll just have to provide clear documentation that this command no longer writes the output into a pipeline variable - as this is now the case, people might still expect it to do this or complain their setup is no longer working after migrating to this module.

Reply:
Removed the writing to the vso output variable and updated the test.

Reply:
Although, I'm not sure what's the purpose of this line:

```powershell
$armAccessPoliciesParameter = [pscustomobject]@{
    list = $armAccessPolicies
}
```

Reply:
If it doesn't add any value, I'll remove this and return the list (`$accessPolicies`) itself as return value.

Reply:
I'll leave it up to @mbraekman

Reply:
> I'll leave it up to @mbraekman

Any thoughts @mbraekman ?

Reply:
The reason for that additional object is to ensure that the output-structure is valid to be used as a parameter-value of an ARM-template.
ARM expects the following structure, which is achieved with the additional conversion:
```
{
  "list": [
     { "tenantId": "123456", ... }
  ]
}
```

While the default structure is:
```
[
  { "tenantId": "123456", ... }
]
```

Reply:
Ok, then I'll leave it like this.